src/test/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessorIT.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Predicate;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+import org.opensearch.client.Response;
+import org.opensearch.neuralsearch.utils.TestHelper;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+
+public class TextEmbeddingProcessorIT extends OpenSearchRestTestCase {
+
+    private static final String indexName = "text_embedding_index";
+
+    private static final String pipelineName = "pipeline-hybrid";
+
+    private final ClassLoader classLoader = this.getClass().getClassLoader();
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public void test_text_embedding_processor() throws IOException {
```

Comment:
[Nit-Pick] : Add @SneakyThrows on the method signature which will clean up the function signature by removing the "throws IOException".

Do it for all the functions in the test.